File: Matlab scripts/testthermal.m
%%This is the test script for thermal analysis 
clear; clc;
latitude=37; %degrees
solarTime=12; %hours
altitude=21; %km
airSpeed=20; %m/stime
totalTime=86400; %%seconds
vent_velocity=0;
[internalHTC,airTemperature,temperatureTrackArray, qnetArray, outCondArray, convectionArray, ...
          absSolarIrradianceArray, radiosityArray, insideConductionArray, ...
          timeArray, payloadTemp, payloadNetPower, powerbrickTemp, powerbrickNetPower, ...
          auterionTemp, auterionNetPower, gpsTemp, gpsNetPower, ethernetTemp, ethernetNetPower, ...
          radioTemp, radioNetPower, battery1Temp, battery1NetPower, battery2Temp, battery2NetPower, ...
          battery3Temp, battery3NetPower, battery4Temp, battery4NetPower, ...
          dTpayload, dTpowerbrick, dTauterion, dTgps, dTethernet, ...
          dTradio, dTbattery1, dTbattery2, dTbattery3, dTbattery4, ...
          dTwallinside, dTwalloutside,TwallinsideTemp, TwallinsideNetPower, TwalloutsideTemp, Qwallout,silvusTemp,silvusNetPower] ...
          = thermalAnalysis(altitude, latitude, 152, solarTime, airSpeed, totalTime,vent_velocity);


%% plotting 


%%solar load 
figure(1)
plot(timeArray,absSolarIrradianceArray)
hold on 
plot(timeArray,outCondArray)
plot(timeArray,-convectionArray)
plot(timeArray,-radiosityArray)
plot(timeArray,Qwallout)
title("heat transfer methods @outside of wall")
xlabel("Time (hrs)")
ylabel("Heat (W)")
legend("Solar Irradiance", "Conduction", "Convection","Radiosity", "Total Heat Flux")
hold off 

%% temperatures of avionics 
figure(2)
plot(timeArray,payloadTemp)
hold on 
plot(timeArray,powerbrickTemp)
plot(timeArray,auterionTemp)
plot(timeArray,gpsTemp)
plot(timeArray,ethernetTemp)
plot(timeArray,radioTemp)
plot(timeArray,silvusTemp)
title("Temperature of each componenet (K)")
xlabel("Time (hrs)")
ylabel("Temperature (K)")
legend("Payload","Power Brick","Auterion","GPS","Ethernet","Radio", "Silvus")
hold off 

%% net power due to radiosity netwrok
figure(3)
plot(timeArray,battery1NetPower)
hold on 
plot(timeArray,battery2NetPower)
plot(timeArray,battery3NetPower)
plot(timeArray,battery4NetPower)
plot(timeArray,TwallinsideNetPower)
title("Heat in/out due to radiation (W)")
xlabel("Time (hrs)")
ylabel("Power (W/m)")
legend("Battery1","Battery2","Battery3","Battery4","Wall Inside")
hold off 

%% plot dT/dt for each component
figure(4)
plot(timeArray,dTpayload);
hold on
plot(timeArray,dTpowerbrick)
plot(timeArray,dTauterion)
plot(timeArray,dTgps)
plot(timeArray,dTethernet)
plot(timeArray,dTradio)
title("Rate of Temperature Change (dT/dt) for Each Component")
xlabel("Time (hrs)")
ylabel("dT/dt [K/s]")
legend("Payload","Power Brick","Auterion","GPS","Ethernet","Radio")
hold off

%% temperature of batteries 
figure(5)
plot(timeArray,battery1Temp)
hold on 
plot(timeArray,battery2Temp)
plot(timeArray,battery3Temp)
plot(timeArray,battery4Temp)
plot(timeArray,TwalloutsideTemp)
plot(timeArray,TwallinsideTemp)
title("Temperature of each battery (K)")
xlabel("Time (hrs)")
ylabel("Temperature (K)")
legend("Battery1","Battery2","Battery3","Battery4","Wall Outside","Wall Inside")
hold off 



%% dT/dt of batteries 
figure(6)
hold on 
plot(timeArray,dTbattery1)
plot(timeArray,dTbattery2)
plot(timeArray,dTbattery3)
plot(timeArray,dTbattery4)
plot(timeArray,dTwallinside)
plot(timeArray,dTwalloutside)
title("dT/dt of each battery (K)")
xlabel("Time (hrs)")
ylabel("dT/dt [K/s]")
legend( "Battery1","Battery2","Battery3","Battery4","Wall Inside","Wall Outside")
hold off

%% net power due to radiosity netwrok
figure(7)
hold on 
plot(timeArray,payloadNetPower)
plot(timeArray,powerbrickNetPower)
plot(timeArray,auterionNetPower)
plot(timeArray,gpsNetPower)
plot(timeArray,ethernetNetPower)
plot(timeArray,radioNetPower)
plot(timeArray,silvusNetPower)
title("Heat in/out due to radiation (W)")
xlabel("Time (hrs)")
ylabel("Power (W/m)")
legend("Payload","Power Brick","Auterion","GPS","Ethernet","Radio","Silvus")
hold off

File: test_thermal.py
import numpy as np
import matplotlib.pyplot as plt
import thermalAnalysis

# Example call to your thermalAnalysis function
# Make sure you implement thermalAnalysis() separately
def main ():
    (
    internalHTC, airTemperature, temperatureTrackArray, qnetArray, outCondArray, convectionArray,
    absSolarIrradianceArray, radiosityArray, insideConductionArray, timeArray,
    payloadTemp, payloadNetPower, powerbrickTemp, powerbrickNetPower,
    auterionTemp, auterionNetPower, gpsTemp, gpsNetPower, ethernetTemp, ethernetNetPower,
    radioTemp, radioNetPower, battery1Temp, battery1NetPower, battery2Temp, battery2NetPower,
    battery3Temp, battery3NetPower, battery4Temp, battery4NetPower,
    dTpayload, dTpowerbrick, dTauterion, dTgps, dTethernet,
    dTradio, dTbattery1, dTbattery2, dTbattery3, dTbattery4,
    dTwallinside, dTwalloutside, TwallinsideTemp, TwallinsideNetPower, TwalloutsideTemp,
    Qwallout, silvusTemp, silvusNetPower
    ) = thermalAnalysis.thermal_analysis(altitude=21, latitude=37, day=152, solarTime=12,
                   airSpeed=20, totalTime=86400, vent_velocity=0)

    # ---- Plotting ---- #

    # Solar load
    plt.figure(1)
    plt.plot(timeArray, absSolarIrradianceArray, label="Solar Irradiance")
    plt.plot(timeArray, outCondArray, label="Conduction")
    plt.plot(timeArray, -convectionArray, label="Convection")
    plt.plot(timeArray, -radiosityArray, label="Radiosity")
    plt.plot(timeArray, Qwallout, label="Total Heat Flux")
    plt.title("Heat transfer methods @outside of wall")
    plt.xlabel("Time (hrs)")
    plt.ylabel("Heat (W)")
    plt.legend()
    plt.grid(True)

    # Temperatures of avionics
    plt.figure(2)
    plt.plot(timeArray, payloadTemp, label="Payload")
    plt.plot(timeArray, powerbrickTemp, label="Power Brick")
    plt.plot(timeArray, auterionTemp, label="Auterion")
    plt.plot(timeArray, gpsTemp, label="GPS")
    plt.plot(timeArray, ethernetTemp, label="Ethernet")
    plt.plot(timeArray, radioTemp, label="Radio")
    plt.plot(timeArray, silvusTemp, label="Silvus")
    plt.title("Temperature of each component (K)")
    plt.xlabel("Time (hrs)")
    plt.ylabel("Temperature (K)")
    plt.legend()
    plt.grid(True)

    # Net power due to radiosity network (batteries & wall inside)
    plt.figure(3)
    plt.plot(timeArray, battery1NetPower, label="Battery1")
    plt.plot(timeArray, battery2NetPower, label="Battery2")
    plt.plot(timeArray, battery3NetPower, label="Battery3")
    plt.plot(timeArray, battery4NetPower, label="Battery4")
    plt.plot(timeArray, TwallinsideNetPower, label="Wall Inside")
    plt.title("Heat in/out due to radiation (W)")
    plt.xlabel("Time (hrs)")
    plt.ylabel("Power (W/m)")
    plt.legend()
    plt.grid(True)

    # dT/dt for each component
    plt.figure(4)
    plt.plot(timeArray, dTpayload, label="Payload")
    plt.plot(timeArray, dTpowerbrick, label="Power Brick")
    plt.plot(timeArray, dTauterion, label="Auterion")
    plt.plot(timeArray, dTgps, label="GPS")
    plt.plot(timeArray, dTethernet, label="Ethernet")
    plt.plot(timeArray, dTradio, label="Radio")
    plt.title("Rate of Temperature Change (dT/dt) for Each Component")
    plt.xlabel("Time (hrs)")
    plt.ylabel("dT/dt [K/s]")
    plt.legend()
    plt.grid(True)

    # Temperature of batteries + walls
    plt.figure(5)
    plt.plot(timeArray, battery1Temp, label="Battery1")
    plt.plot(timeArray, battery2Temp, label="Battery2")
    plt.plot(timeArray, battery3Temp, label="Battery3")
    plt.plot(timeArray, battery4Temp, label="Battery4")
    plt.plot(timeArray, TwalloutsideTemp, label="Wall Outside")
    plt.plot(timeArray, TwallinsideTemp, label="Wall Inside")
    plt.title("Temperature of each battery (K)")
    plt.xlabel("Time (hrs)")
    plt.ylabel("Temperature (K)")
    plt.legend()
    plt.grid(True)

    # dT/dt of batteries + walls
    plt.figure(6)
    plt.plot(timeArray, dTbattery1, label="Battery1")
    plt.plot(timeArray, dTbattery2, label="Battery2")
    plt.plot(timeArray, dTbattery3, label="Battery3")
    plt.plot(timeArray, dTbattery4, label="Battery4")
    plt.plot(timeArray, dTwallinside, label="Wall Inside")
    plt.plot(timeArray, dTwalloutside, label="Wall Outside")
    plt.title("dT/dt of each battery (K)")
    plt.xlabel("Time (hrs)")
    plt.ylabel("dT/dt [K/s]")
    plt.legend()
    plt.grid(True)

    # Net power due to radiosity (all avionics + silvus)
    plt.figure(7)
    plt.plot(timeArray, payloadNetPower, label="Payload")
    plt.plot(timeArray, powerbrickNetPower, label="Power Brick")
    plt.plot(timeArray, auterionNetPower, label="Auterion")
    plt.plot(timeArray, gpsNetPower, label="GPS")
    plt.plot(timeArray, ethernetNetPower, label="Ethernet")
    plt.plot(timeArray, radioNetPower, label="Radio")
    plt.plot(timeArray, silvusNetPower, label="Silvus")
    plt.title("Heat in/out due to radiation (W)")
    plt.xlabel("Time (hrs)")
    plt.ylabel("Power (W/m)")
    plt.legend()
    plt.grid(True)

    plt.show()

# This makes sure the script only runs when you execute it directly
if __name__ == "__main__":
    main()